File: algorithms/R/Range Sum of Sorted Subarray Sums/Range Sum of Sorted Subarray Sums.cpp
class Solution {
public:
    int mod = 1e9 + 7;
    int rangeSum(vector<int>& nums, int n, int left, int right) {
        vector<int> ans;
        for(int i = 0; i < n; i++){
            int psum = 0;
            for(int j = i; j < n; j++){
                psum += nums[j];
                ans.push_back(psum);
            }
        }
        sort(ans.begin(), ans.end());
        int res_sum = 0;
        for(int i = left - 1; i < right; i++){
            res_sum =  (res_sum +  ans[i]) % mod;
        }
        return res_sum;
    }
};

File: scripts/algorithms/R/Range Sum of Sorted Subarray Sums/Range Sum of Sorted Subarray Sums.py
# Runtime: 276 ms (Top 95.45%) | Memory: 37.2 MB (Top 59.09%)
from itertools import accumulate

class Solution:
    def rangeSum(self, nums, n, left, right):
        acc = []

        for i in range(n):
            acc.extend(accumulate(nums[i:]))

        acc.sort()

        return sum(acc[left - 1:right]) % (10**9 + 7)